daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Decimal.scala

```diff
@@ -72,14 +72,14 @@ object Decimal {
             .setScale(scale))
     })
 
+  private val hasExpectedFormat: String => Boolean =
+    """[+-]?\d{1,28}(\.\d{1,10})?""".r.pattern.matcher(_).matches()
```

Comment:
Unfortunately if you don't do this the whole expression goes in the lambda body:

```suggestion
    {
      val p = """[+-]?\d{1,28}(\.\d{1,10})?""".r.pattern
      p.matcher(_).matches()
    }
```

```scala
scala> reify({
     |       val p = """[+-]?\d{1,28}(\.\d{1,10})?""".r.pattern
     |       p.matcher(_).matches()
     | }: String => Boolean)
res1: reflect.runtime.universe.Expr[String => Boolean] =
Expr[String => Boolean](({
  val p = Predef.augmentString("[+-]?\\d{1,28}(\\.\\d{1,10})?").r.pattern;
  ((x$1) => p.matcher(x$1).matches())
}: Function1[Predef.String, Boolean]))

reify(("""[+-]?\d{1,28}(\.\d{1,10})?""".r.pattern.matcher(_).matches()): String => Boolean)
res5: reflect.runtime.universe.Expr[String => Boolean] = Expr[String => Boolean]((((x$1) => Predef.augmentString("[+-]?\\d{1,28}(\\.\\d{1,10})?").r.pattern.matcher(x$1).matches()): Function1[Predef.String, Boolean]))
```

Reply:
Thx.